tools/test.py

```diff
@@ -126,6 +126,12 @@
         all_tests = {}
         tests = {}
 
+        # Add the test macro, when compiling all test
+        if options.macros is None:
+           options.macros = (['MBED_TEST'])
+        else:
+            options.macros.extend(['MBED_TEST'])
```

Comment:
Please rewrite as:
```python
options.macros.append('MBED_TEST')
```

Reply:
Fixed it on dependent PR https://github.com/ARMmbed/mbed-os/pull/7627